service/operator/service.go

```diff
@@ -164,3 +165,12 @@ func (s *Service) deleteFunc(obj interface{}) {
 		s.logger.Log("error", fmt.Sprintf("%#v", err), "event", "delete")
 	}
 }
+
+func (s *Service) updateFunc(oldObj interface{}, newObj interface{}) {
+	s.logger.Log("debug", "executing the operator's updateFunc")
```

Comment:
FYI we lock these functions in other operators with a mutex. See for instance https://github.com/giantswarm/draughtsman-operator/blob/f3e8eaf183176479fed6e5228b2611dec1b5ee9d/service/operator/service.go#L141.

Reply:
Ack. I wasn't sure if the locking was needed because the state is in the Azure API. But I think it makes sense to lock. I'll fix.

Reply:
Locking with the mutex is added.